### PR TITLE
Remove next/link legacy behavior

### PR DESCRIPTION
**Pull request by johnnyomair**
This PR only demonstrates the necessary code changes. I will add a migration guide and changesets once we decide we want to change that now.

Changes necessary in projects:

Either: 
- All custom link blocks need to render a HTML tag (a or span) that accepts a className
- All link stylings need to be change to style the link block, not the a-tag in the link block
- className needs to be prop drilled in LinkBlock

This will require lots of work in the projects I suppose. The [Next codemod ](https://nextjs.org/docs/app/building-your-application/upgrading/codemods#remove-a-tags-from-link-components) will not be of much help here.

Or: Add `legacyBehavior` prop to link block usages.

---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-803

### CODE REVIEW

File: demo/site/src/header/Header.tsx [hunk 1]
```diff
@@ -66,14 +66,18 @@ const TopLevelLinkContainer = styled.li`
...
-    color: ${({ $active, theme }) => ($active ? theme.colors.primary : theme.colors.black)};
+    color: ${({ theme }) => theme.colors.black};
 
     &:hover {
         text-decoration: underline;
     }
+
+    &.active {
+        color: ${({ theme }) => theme.colors.primary};
+    }
```
Comment (johnnyomair) [inline demo/site/src/header/Header.tsx:80]: I achieved the active styling using a plain old class name that's set on the PageLink component. Another solution would be to remove the active check in the component and add an `active` prop that's set from outside. What do you prefer?

Comment (johnnyomair) [reply demo/site/src/header/Header.tsx:84]: Add activeClassName as discussed.

**Comment by johnnyomair** [comment]
@nsams @thomasdax98 I've added the changes we discussed yesterday, so this PR is ready to go.

File: docs/docs/migration/migration-from-v6-to-v7.md [hunk 2]
```diff
@@ -831,6 +831,53 @@ The `InternalLinkBlock` provided by `@comet/cms-site` is deprecated.
 Instead, implement your own `InternalLinkBlock`.
 This is needed for more flexibility, e.g., support for internationalized routing.
 
+### Add `legacyBevavior` to all link block usages
+
+Previously, Next required the `Link` component to have a child `<a>` tag. To style this tag correctly in the application, none of the library link blocks rendered the tag, but cloned the children with the correct props instead. However, since Next v13 the `Link` component no longer requires a child `<a>` tag. Consequently, we don't need to render the tag for the `InternalLinkBlock` (which uses `Link` internally) anymore. In order to style all link blocks correctly, we now render an `<a>` tag for all other link blocks.
```
Comment (nsams) [inline docs/docs/migration/migration-from-v6-to-v7.md:836]: that paragraph describes the situation without legacyBehaviour but in the migration guide we advise to use the legacyBehaviour

Comment (johnnyomair) [reply docs/docs/migration/migration-from-v6-to-v7.md:836]: I wanted to give some context why this change is necessary. But I'll rewrite it to make it more specific.

Comment (johnnyomair) [reply docs/docs/migration/migration-from-v6-to-v7.md]: Changed: fc7aa4e95f07d7e25d94be477b586c37258164fe

**Review by thomasdax98 — DISMISSED** [review]

File: .changeset/thirty-countries-refuse.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+"@comet/cms-site": major
+---
+
+Remove `next/link` legacy behavior
```
Comment (nsams) [inline .changeset/thirty-countries-refuse.md:5]:
```suggestion
Remove `next/link` legacy behavior as default behavior
```

**Review by nsams — DISMISSED** [review]

**Review by thomasdax98 — APPROVED** [review]

**Pull request merged**